test/utils.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+const { platform } = require('os');
+
+/**
+ * @typedef {object} Logger
+ * @property {(...message: unknown[]) => void} error
+ */
+
+/** @type {{readonly [char: string]: string}} */
+const INVISIBLES_MAP = Object.freeze(
+  Object.assign(Object.create(null), {
+    '\0': '\\0', // ␀ (0x00)
+    '\b': '\\b', // ␈ (0x08)
+    '\t': '\\t', // ␉ (0x09)
+    '\n': '\\n', // ␊ (0x0A)
+    '\v': '\\v', // ␋ (0x0B)
+    '\f': '\\f', // ␌ (0x0C)
+    '\r': '\\r', // ␍ (0x0D)
+  })
+);
+const INVISIBLES_REGEXP = /[\0\x08-\x0D]/g;
+
+/** Used to check if the process is running in a CI environment. */
+const IS_CI = process.env.CI && String(process.env.CI).toLowerCase() === 'true';
+
+/** Determines if the OS is Windows */
+const IS_WINDOWS = platform() === 'win32';
+
+/**
+ * @param {string} str
+ */
+function escapeInvisibles(str) {
```

Comment:
It would have been better if a re-implementation of this function would have been done in a separate PR.